workspace_definitions.bzl

```diff
@@ -45,8 +54,4 @@ def rules_foreign_cc_dependencies(
 
     native.register_toolchains(*native_tools_toolchains)
     if register_default_tools:
-        native.register_toolchains(
-            "@rules_foreign_cc//tools/build_defs:preinstalled_cmake_toolchain",
-            "@rules_foreign_cc//tools/build_defs:preinstalled_ninja_toolchain",
-            "@rules_foreign_cc//tools/build_defs:preinstalled_make_toolchain",
-        )
+        prebuilt_toolchains(cmake_version, ninja_version)
```

Comment:
I think you also need to call `preinstalled_toolchains` here so that if there isn't a matching exec constraint for the platform e.g. ninja on aarch64 then we at least try to fallback to the preinstalled version. This should probably also be a parameter to the repo macro such as `use_preinstalled_tools`

Reply:
Added an option to register preinstalled toolchains